pkgs/os-specific/linux/firmware/raspberrypi-wireless/default.nix

```diff
@@ -1,52 +1,49 @@
-{ stdenv, fetchurl, dpkg }:
+{ stdenv, fetchurl, fetchFromGitHub, dpkg }:
 
 stdenv.mkDerivation rec {
   name = "raspberrypi-wireless-firmware-${version}";
-  version = "2018-05-30";
+  version = "2018-08-20";
 
   srcs = [
-    (fetchurl {
-      url = "https://archive.raspberrypi.org/debian/pool/main/b/bluez-firmware/bluez-firmware_1.2-3+rpt5.debian.tar.xz";
-      sha256 = "06zpyrz6frkgjy26hr3998klnhjdqxwashgjgvj9rgbcqy70nkxg";
+    (fetchFromGitHub {
+      name = "bluez-firmware";
+      owner = "RPi-Distro";
+      repo = "bluez-firmware";
+      rev = "ade2bae1aaaebede09abb8fb546f767a0e4c7804";
+      sha256 = "07gm76gxp5anv6paryvxcp34a86fkny8kdlzqhzcpfczzglkp6ag";
     })
-    (fetchurl {
-      url = "https://archive.raspberrypi.org/debian/pool/main/f/firmware-nonfree/firmware-brcm80211_20161130-3+rpt3_all.deb";
-      sha256 = "10l74ac28baprnsiylf2vy4pkxgb3crixid90ngs6si9smm7rn6z";
+    (fetchFromGitHub {
+      name = "firmware-nonfree";
+      owner = "RPi-Distro";
+      repo = "firmware-nonfree";
+      rev = "b518de45ced519e8f7a499f4778100173402ae43";
+      sha256 = "1d5026ic9awji6c67irpwsxpxgsc0dhn11d3abkxi2vvra1pir4g";
     })
   ];
 
   sourceRoot = ".";
-  dontBuild = true;
-  # Firmware blobs do not need fixing and should not be modified
-  dontFixup = true;
 
-
-  # Unpack the debian package
-  nativeBuildInputs = [ dpkg ];
-  unpackCmd = ''
-    if ! [[ "$curSrc" =~ \.deb$ ]]; then return 1; fi
-    dpkg -x "$curSrc" .
-  '';
+  phases = [ "unpackPhase" "installPhase" ];
 
   installPhase = ''
     mkdir -p "$out/lib/firmware/brcm"
 
     # Wifi firmware
-    for filename in lib/firmware/brcm/brcmfmac434??-sdio.*; do
+    for filename in firmware-nonfree/brcm/brcmfmac434??-sdio.*; do
       cp "$filename" "$out/lib/firmware/brcm"
     done
 
     # Bluetooth firmware
-    cp broadcom/*.hcd "$out/lib/firmware/brcm"
+    cp bluez-firmware/broadcom/*.hcd "$out/lib/firmware/brcm"
   '';
 
   outputHashMode = "recursive";
   outputHashAlgo = "sha256";
-  outputHash = "1gwzasl5w5nc0awqv3w2081ns63wd1yds0xh0dg95dc6brnqhhf8";
+  outputHash = "1s5gb00v42s5izbaw8irs1fwvhh7z9wl07czc0nkw6p91871ivb7";
```

Comment:
Does this really need to be a fixed-output derivation? There something else fetched other than the declared `srcs`, right?

Reply:
That was done to prevent it from needing to be cross-compiled. See #44349